src/rest_vol.c

```diff
@@ -2761,6 +2768,68 @@ RV_convert_dataspace_shape_to_JSON(hid_t space_id, char **shape_body, char **max
     return ret_value;
 } /* end RV_convert_dataspace_shape_to_JSON() */
 
+/* Helper funciton to store the version of the external HSDS server */
+herr_t
+RV_parse_HSDS_version(char *HTTP_response)
+{
+    yajl_val parse_tree            = NULL, key_obj;
+    herr_t   ret_value             = SUCCEED;
+    char    *version_response      = NULL;
+    char    *version_field         = NULL;
+    int      numeric_version_field = 0;
+
+    /* Parse information from response */
+#ifdef RV_CONNECTOR_DEBUG
+    printf("-> Retrieving HSDS version from server's HTTP response\n\n");
+#endif
+
+    if (!HTTP_response)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "HTTP response buffer was NULL");
+
+    if (NULL == (parse_tree = yajl_tree_parse(HTTP_response, NULL, 0)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsing JSON failed");
+
+    /* Retrieve version */
+    if (NULL == (key_obj = yajl_tree_get(parse_tree, hsds_version_keys, yajl_t_string)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "failed to parse HSDS version");
+
+    if (!YAJL_IS_STRING(key_obj))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "parsed HSDS version is not a string");
+
+    if (NULL == (version_response = YAJL_GET_STRING(key_obj)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "HSDS version was NULL");
+
+    /* Parse HSDS version into struct */
+    if (NULL == (version_field = strtok_r(version_response, ".", &version_response)))
```

Comment:
Using strtok_r is reasonable here for the future, but you should use a separate variable for its third argument for storing state rather than reusing a pointer to the same pointer.